### ISSUE TITLE
Problem: impossible to run tests in a different database

### ISSUE THREAD

**Issue by yrashk** [omnigres/omnigres#351]
In some obscure cases (like omni_ext), we want to be able to test against multiple databases in the same instance. For example, some extensions may have database-specific results.

Currently, we can't do `create database` because we're wrapping everything into a transaction (`CREATE DATABASE cannot run inside a transaction block` error), and we can't switch databases on the fly (pg_yregress doesn't have support for it.

To address this, we can perhaps add the following functionality:

## Non-transactional tests in pg_yregress

Something like this?

```yaml
- query: create database another_db
   transaction: false
```

## Ability to send queries to other databases in pg_yregress

Syntax-wise this can be as simple as:

```yaml
- query: select omni_ext_test.alloc_shmem_database_local()
   database: another_db
```

We would need to connect to manage another connection. This may not be too bad, but we need to think about connection reuse (may be not in the initial draft, though?)

_Originally posted by @yrashk in https://github.com/omnigres/omnigres/pull/278#discussion_r1383375407_

**Bounty size: $100**

[Read more about bounties](https://github.com/omnigres/omnigres/wiki/Bounties)

---

<details>
<summary><b>Don't want to start from scratch?</b></summary>
<br>

A change for kickstarting the process:

```diff
diff --git a/pg_yregress/pg_yregress.c b/pg_yregress/pg_yregress.c
index 481666ac0..74e1c9f80 100644
--- a/pg_yregress/pg_yregress.c
+++ b/pg_yregress/pg_yregress.c
@@ -54,6 +54,21 @@ static bool populate_ytest_from_fy_node(struct fy_document *fyd, struct fy_node
       }
     }
 
+    {
+      // Is the test meant to be ran on a different database?
+      struct fy_node *database = fy_node_mapping_lookup_by_string(test, STRLIT("database"));
+      y_test->database.base = NULL;
+
+      if (database != NULL) {
+        if (!fy_node_is_scalar(database)) {
+          fprintf(stderr, "database should be a string, got: %s",
+                  fy_emit_node_to_string(database, FYECF_DEFAULT));
+          return false;
+        }
+        y_test->database.base = fy_node_get_scalar(database, &y_test->database.len);
+      }
+    }
+
     {
       // Is the test meant to be negative? As in "if test succeeds, it's a failure"
       struct fy_node *negative = fy_node_mapping_lookup_by_string(test, STRLIT("negative"));
diff --git a/pg_yregress/pg_yregress.h b/pg_yregress/pg_yregress.h
index 749fdbb17..301a58cb1 100644
--- a/pg_yregress/pg_yregress.h
+++ b/pg_yregress/pg_yregress.h
@@ -101,6 +101,7 @@ typedef struct {
   struct fy_node *node;
   bool commit;
   bool negative;
+  iovec_t database;
 } ytest;
 
 bool ytest_run(ytest *test);
diff --git a/pg_yregress/test.c b/pg_yregress/test.c
index f3ac1a8ac..4e5862f04 100644
--- a/pg_yregress/test.c
+++ b/pg_yregress/test.c
@@ -94,6 +94,9 @@ proceed:
 
   assert(test->instance != NULL);
   PGconn *conn;
+  if (test->database.base != 0) {
+  }
+
   if (default_conn == NULL) {
     struct fy_node *test_container = fy_node_get_parent(test->node);
     assert(fy_node_is_sequence(test_container));
diff --git a/pg_yregress/test.yml b/pg_yregress/test.yml
index 1b2b9c50b..f5351bcc4 100644
--- a/pg_yregress/test.yml
+++ b/pg_yregress/test.yml
@@ -5,6 +5,8 @@ name: pg_yregress
 instances:
   main:
     default: true
+    init:
+    - create database another_db
   restarted:
     init:
     # This will be tested
@@ -410,4 +412,11 @@ tests:
     netmask: null
     auth_method: trust
     options: null
-    error: null
\ No newline at end of file
+    error: null
+
+- name: another database
+  database: another_database
+  query: |
+    select *
+    from
+        pg_class
\ No newline at end of file
```
</details>

**Comment by idoqo** [comment]
@yrashk considering that we typically want to run other queries after the `create database...` query, I'm thinking the option to send queries to another database makes more sense.

Then the `create database` call can stay in the init block.

Wdyt?

**Comment by yrashk** [comment]
Implemented in #399
